File: New_NativeUI/Panels/UIMenuColorPanel.cs
using System.Collections.Generic;
using System.Drawing;
using CitizenFX.Core.Native;
using CitizenFX.Core.UI;
using CitizenFX.Core;
using System.Threading.Tasks;
using System;

namespace ScaleformUI
{
	public enum ColorPanelType { Hair, Makeup }
	public class UIMenuColorPanel : UIMenuPanel
	{
		public string Title { get; set; }
		public ColorPanelType ColorPanelColorType { get; set; }
		internal int _value { get; set; }
		public event ColorPanelChangedEvent OnColorPanelChange;
		public int CurrentSelection
		{
			get
			{
				//_getValue();
				return _value;
			}
			set
			{
				_value = value;
				if (value > 63)
					_value -= 63;
				if (value < 0)
					_value += 63;
				_setValue(value);
			}
		}
		public UIMenuColorPanel(string title, ColorPanelType ColorType, int startIndex = 0)
		{
			Title = title??"Color Panel";
			ColorPanelColorType = ColorType;
			_value = startIndex;
		}

		internal void PanelChanged()
		{
			OnColorPanelChange?.Invoke(ParentItem, this, CurrentSelection);
		}

		/*
		private void //UpdateSelection(bool update)
		{
			if (update)
			{
				ParentItem.Parent.ListChange(ParentItem, ParentItem.Index);
				ParentItem.ListChangedTrigger(ParentItem.Index);
			}
		}*/

		public async void _getValue()
        {
			var it = this.ParentItem.Parent.MenuItems.IndexOf(this.ParentItem);
			var van = this.ParentItem.Panels.IndexOf(this);
			API.BeginScaleformMovieMethod(ScaleformUI._ui.Handle, "GET_VALUE_FROM_PANEL");
			API.ScaleformMovieMethodAddParamInt(it);
			API.ScaleformMovieMethodAddParamInt(van);
			var ret = API.EndScaleformMovieMethodReturnValue();
			while (!API.IsScaleformMovieMethodReturnValueReady(ret)) await BaseScript.Delay(0);
			_value = API.GetScaleformMovieMethodReturnValueInt(ret);
		}

		public void _setValue(int val)
        {
			var it = ParentItem.Parent.MenuItems.IndexOf(this.ParentItem);
			var van = ParentItem.Panels.IndexOf(this);
			ScaleformUI._ui.CallFunction("SET_COLOR_PANEL_VALUE", it, van, val);
		}
	}
}

File: ScaleformUI_Lua/src/panels/UIMenuColorPanel.lua
UIMenuColorPanel = setmetatable({}, UIMenuColorPanel)
UIMenuColorPanel.__index = UIMenuColorPanel
UIMenuColorPanel.__call = function() return "UIMenuPanel", "UIMenuColorPanel" end

---New
---@param title string
---@param colorType int
---@param startIndex number
function UIMenuColorPanel.New(title, colorType, startIndex, colors)
	if colors ~= nil then
		colorType = 2
	end
	
	local _UIMenuColorPanel = {
		Title = title or "Color Panel",
		ColorPanelColorType = colorType,
		value = startIndex or 0,
		CustomColors = colors or nil,
		ParentItem = nil, -- required
		OnColorPanelChanged = function(item, panel, newindex) end
	}
	return setmetatable(_UIMenuColorPanel, UIMenuColorPanel)
end

---SetParentItem
---@param Item table
function UIMenuColorPanel:SetParentItem(Item) -- required
	if not Item() == nil then
		self.ParentItem = Item
	else
		return self.ParentItem
	end
end

function UIMenuColorPanel:CurrentSelection(new_value)
	if new_value ~= nil then
		self.value = new_value
		if self.ParentItem ~= nil and self.ParentItem:SetParentMenu() ~= nil and self.ParentItem:SetParentMenu():Visible() then
			local it = IndexOf(self.ParentItem:SetParentMenu().Items, self.ParentItem)
			local van = IndexOf(self.ParentItem.Panels, self)
			ScaleformUI.Scaleforms._ui:CallFunction("SET_COLOR_PANEL_VALUE", false, it, van, new_value)
		end
	else
		return self.value
	end
end